core/slice.go

```diff
@@ -286,6 +295,37 @@ func (sl *Slice) backfillPETXs(header *types.Header, subManifest types.BlockMani
 	}
 }
 
+func (sl *Slice) computeTps(block *types.Block) uint32 {
+	var tps uint32
+	if common.NodeLocation.Context() != common.ZONE_CTX {
+		for _, val := range sl.subTps {
+			tps += val
+		}
+
+	} else {
+		sl.tpsHistory.Add(block.Hash(), len(block.Transactions()))
```

Comment:
I think we should use a simple IIR. Replace `sl.tpsHistory` cache with an atomic `sl.tps` value, and the entire 'else' block could/should reduce to:
```
} else {
    dt := block.Time() - parent.Time()             // Calculate time delta
    instantTps := len(block.Transactions()) / dt   // Calculate instantaneous TPS
    sl.tps = alpha * sl.tps + instantTps           // Update EWMA with new TPS measurement.
}
```
Where `alpha` is a constant we can use to tune the EWMA filter's responsiveness.

Reply:
Note: this is just a suggested improvement. Not a bug.